### PR TITLE
fix: Fix examples not compiling due to function rename

### PR DESCRIPTION
**Pull request by lszetg**
# Pull Request Template

## Motivation and Context

The examples didn't compile, they we'rent updated to reflect the changes made during the transition to Networks.

## Summary

Provide a short summary of changes in this pull request.

- removed `setCurrency` 
- updated binding examples (moved `get/set Network` to specific examples that require it)
- examples use networks fetched form the backend
- some files might have been reformatted (we're using the same formatter so left them that way)
- 
## Checklist

Please ensure that your PR meets the following requirements:

### General

- [ ] Code follows project style and guidelines.
- [ ] No redundant or duplicate code.
- [ ] No added new dependencies without clear justification and approval.

### Documentation

- [ ] Added/updated relevant documentation (e.g., README, inline comments).
- [ ] Added a CHANGELOG entry for major changes.

### Testing

- [ ] Added relevant test cases, leveraging tools like `rstest` or similar for reusable tests.
- [ ] Tests pass locally.

### Build & CI/CD

- [ ] Confirmed no CI/CD pipeline issues.
- [ ] Updated build scripts where necessary.

## Comments to Reviewer

Please provide any comments or points of consideration for the reviewers, such as specific areas of focus, potential edge cases, or context that would aid the review process.

### Reviewer Responsibilities

- [ ] Code adheres to style and guidelines.
- [ ] All tests are appropriately covered and they pass.
- [ ] Consider performance and/or security impacts.
- [ ] Constructive feedback is provided.

### CODE REVIEW

File: sdk/bindings/android/tests/src/main/java/com/cawaena/examples/CreatePurchaseRequest08.java [hunk 1]
```diff
@@ -1,5 +1,10 @@
 package com.cawaena.examples;
 
+import com.cawaena.Wallet;import com.cawaena.model.Network;
```
Comment (antbern) [inline sdk/bindings/android/tests/src/main/java/com/cawaena/examples/CreatePurchaseRequest08.java:3]: Fix formatting:
```suggestion
import com.cawaena.Wallet;
import com.cawaena.model.Network;
```

File: sdk/bindings/android/tests/src/main/java/com/cawaena/examples/SendCompliment20.java [hunk 2]
```diff
@@ -31,6 +34,21 @@ public static void main(String[] args) throws InterruptedException, ExecutionExc
...
+            try {
+                ObjectMapper objectMapper = new ObjectMapper();
+                networksList = objectMapper.readValue(networks, new TypeReference<List<Network>>() {});
+            } catch (JsonProcessingException e) {
+                throw new RuntimeException("Error processing JSON response", e);
+            }
+
+            Network iotaNetwork = networksList.get(0);
+            // set the network configuration for the wallet
+            sdk.setNetwork(iotaNetwork.id);
```
Comment (antbern) [inline sdk/bindings/android/tests/src/main/java/com/cawaena/examples/SendCompliment20.java:51]: Can we not just move this to `utils.java` ? Feels very duplicated....

File: sdk/bindings/wasm/examples/07-get_balance.ts [hunk 3]
```diff
@@ -14,6 +14,11 @@ async function main() {
 
     await sdk.createWalletFromMnemonic(pin, mnemonic);
 
+    // fetch networks from backend
+    let networks = await sdk.getNetworks();
+    // set the network configuration for the wallet
+    sdk.setNetwork(networks[0].id);
+
```
Comment (antbern) [inline sdk/bindings/wasm/examples/07-get_balance.ts:21]: same, can we put it in `utils.ts` ? 🌵

File: sdk/examples/14_get_exchange_rate.rs [hunk 4]
```diff
@@ -13,6 +13,11 @@ async fn main() {
     sdk.create_new_user(&user.username).await.unwrap();
     sdk.init_user(&user.username).await.unwrap();
 
+    // Fetch networks from backend
+    let networks = sdk.get_networks().await.unwrap();
+    let iota_network_id = &networks.first().unwrap().id;
+    sdk.set_network(iota_network_id.to_string()).await.unwrap();
+
```
Comment (antbern) [inline sdk/examples/14_get_exchange_rate.rs:20]: Here too?

File: sdk/examples/utils.rs [hunk 5]
```diff
@@ -39,37 +36,6 @@ pub async fn init_sdk() -> (Sdk, CleanUp) {
...
-            network_type: NetworkType::Stardust {
-                node_urls: vec![String::from("https://api.testnet.iotaledger.net")],
-            },
-        },
-        Network {
-            id: String::from("67a2080ddf55756bae21e7f5"),
-            name: String::from("Eth Sepolia"),
-            currency: String::from("ETH"),
-            block_explorer_url: String::from("https://sepolia.explorer.mode.network"),
-            enabled: true,
```
Comment (kristifidani) [inline sdk/examples/utils.rs:60]: we can set the networks in the untils instead to avoid some repetition 😄

Comment (lszetg) [reply sdk/examples/utils.rs:60]: https://github.com/ETOSPHERES-Labs/cawaena-sdk/pull/56#discussion_r1995624426
:D that's why it is not in utils, but we can have a discussion

File: sdk/bindings/android/tests/src/main/java/com/cawaena/examples/SendCompliment20.java (hunk 2 above)
Comment (kristifidani) [reply sdk/bindings/android/tests/src/main/java/com/cawaena/examples/SendCompliment20.java:51]: Well yeah that is true. That would be the case also for the access token. Not every example needs it (e.g. create new user example) but we still generate it in the `utils` since it is easier to change / update in one place rather than going through each example.  

The idea behind the utils in this case is to have all the sdk configuration / setup inside the utils for simplicity and the networks is part of this configuration. I believe we are going to do some updates on the Networks and it would be easier to have it in one place :D

Comment (lszetg) [reply sdk/bindings/android/tests/src/main/java/com/cawaena/examples/SendCompliment20.java:51]: hm, to move `set/get network` to backend, we need to remove `x-username` (sdk + backend) from `get_networks` request to make it work 

```rust
 async fn get_networks_backend(&self) -> Result<Vec<Network>> {
        let config = self.config.as_ref().ok_or(crate::Error::MissingConfig)?;
        let username = &self
            .active_user
            .as_ref()
            .ok_or(crate::Error::UserNotInitialized)?
            .username;
        let access_token = self
            .access_token
            .as_ref()
            .ok_or(crate::error::Error::MissingAccessToken)?;
        let backend_networks = get_networks(config, username, access_token).await?;
        let networks: Vec<Network> = backend_networks.iter().map(|n| Network::from(n.clone())).collect();

        Ok(networks)
    }
```

downside is that we're _counting active users_ and we would miss these calls

Comment (antbern) [reply sdk/bindings/android/tests/src/main/java/com/cawaena/examples/SendCompliment20.java:51]: That is not what we meant with these comments. We simply meant to move the call to `setNetwork` ( and the related code to get an id) to utils. Nothing about moving set/get network to backend 🌵

Comment (lszetg) [reply sdk/bindings/android/tests/src/main/java/com/cawaena/examples/SendCompliment20.java:51]: just updated my comment :D

Comment (lszetg) [reply sdk/bindings/android/tests/src/main/java/com/cawaena/examples/SendCompliment20.java:51]: friday text editing is... tricky

Comment (antbern) [reply sdk/bindings/android/tests/src/main/java/com/cawaena/examples/SendCompliment20.java:51]: Alright yes now I understand 🤣 My bad. Then let's leave it like this for now and we can work on removing that header (we can take the username directly from the token claims instead, but this is another ticket) 💯

**Review by kristifidani — APPROVED** [review]
Lets go 🚀

**Review by antbern — APPROVED** [review]
